CAN-D/Src/can.c

```diff
@@ -118,14 +120,20 @@ void HAL_CAN_RxFifo0MsgPendingCallback(CAN_HandleTypeDef* canHandle)
     Error_Handler();
   }
 
-  // Pass along CAN RX data depending on the current configuration
+  // Check if SD Logging is enabled
   if (mAppConfiguration.SDStorage == APP_ENABLE)
   {
     // Send CAN data to the SD Card via SPI
   }
-  if (mAppConfiguration.USBTransfer == APP_ENABLE)
+
+  // Check if USB Streaming is enabled in the application and if a USB device 
+  // is connected to this application
+  if (mAppConfiguration.USBStream == APP_ENABLE && hUsbDeviceFS.dev_state == USBD_STATE_CONFIGURED)

```

Comment:
Should this configuration check be here in the CAN RX callback before populating the queues (osMessagePut())?
Or should we just populate all queues with the data and check the configuration in the task thread (i.e. in bridge.c) before pending the queues?

Reply:
I think this is the better way to do it. Don't post it in the queue if USB isnt enabled so that the task thread doesn't have to wake up just to check the configuration